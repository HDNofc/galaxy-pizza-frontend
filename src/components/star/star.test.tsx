import { render } from '@testing-library/react';
import Star from './star';

describe('<Star>', () => {
  it('отображает корректные svg в зависимости от переданного пропса `type`', () => {
    const { container, rerender } = render(<Star />);
    expect(container.firstChild).toMatchSnapshot();

    rerender(<Star type="full" />);
    expect(container.firstChild).toMatchSnapshot();

    rerender(<Star type="half" />);
    expect(container.firstChild).toMatchSnapshot();

    rerender(<Star type="zero" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
