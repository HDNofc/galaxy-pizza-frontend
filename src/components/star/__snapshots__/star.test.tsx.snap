// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star> отображает корректные svg в зависимости от переданного пропса \`type\` 1`] = `
<div
  class="Star-sc-16jz84-0 gjzZhz"
>
  <svg
    style="height: 20px;"
  >
    full.svg
  </svg>
</div>
`;

exports[`<Star> отображает корректные svg в зависимости от переданного пропса \`type\` 2`] = `
<div
  class="Star-sc-16jz84-0 gjzZhz"
>
  <svg
    style="height: 20px;"
  >
    full.svg
  </svg>
</div>
`;

exports[`<Star> отображает корректные svg в зависимости от переданного пропса \`type\` 3`] = `
<div
  class="Star-sc-16jz84-0 gjzZhz"
>
  <svg
    style="height: 20px;"
  >
    half.svg
  </svg>
</div>
`;

exports[`<Star> отображает корректные svg в зависимости от переданного пропса \`type\` 4`] = `
<div
  class="Star-sc-16jz84-0 gjzZhz"
>
  <svg
    style="height: 20px;"
  >
    zero.svg
  </svg>
</div>
`;
